Bond/Extensions/OSX/NSTableView+Bond.swift

```diff
@@ -0,0 +1,138 @@
+//
+//  The MIT License (MIT)
+//
+//  Copyright (c) 2015 Michail Pishchagin (@mblsha)
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Cocoa
+
+public protocol BNDTableViewDelegate {
+  func createCell(row: Int, array: AnyObject, tableView: NSTableView) -> NSTableCellView
```

Comment:
Delegate protocol should be typed, like this:

```swift
public protocol BNDTableViewDelegate {
  typealias Element
  func createCell(row: Int, array: ObservableArray<Element>, tableView: NSTableView) -> NSTableCellView
}
```

Then, you should generalize BNDTableViewDataSource over the delegate type or order to have the delegate as a property:

```swift
class BNDTableViewDataSource<DelegateType: BNDTableViewDelegate>: NSObject, NSTableViewDataSource, NSTableViewDelegate {
  ...
  private var delegate: DelegateType?
  ...
}

Next, you'll be able to create an initializer like:

```swift
init(array: ObservableArray<DelegateType.Element>, tableView: NSTableView, delegate: DelegateType) {
  self.delegate = delegate
  ...
}
```

Actually element type you can get from delegate type: DelegateType.Element. Hope this helps.